optimum/habana/utils.py

```diff
@@ -415,3 +416,37 @@ def get_device_count():
         return htexp.hpu.device_count()
     else:
         raise ValueError("No hpu is found avail on this system")
+
+class SetTrueOrFalseOrNone(argparse.Action):
```

Comment:
Since this is only used in the text-generation example, I think this should rather go in https://github.com/huggingface/optimum-habana/blob/main/examples/text-generation/utils.py